src/zenml/integrations/label_studio/steps/label_studio_standard_steps.py

```diff
@@ -0,0 +1,272 @@
+            storage.
+        google_application_credentials: Specify the Google application
+            credentials to use for the storage.
+        aws_access_key_id: Specify the AWS access key ID to use for the
+            storage.
+        aws_secret_access_key: Specify the AWS secret access key to use for the
+            storage.
+        aws_session_token: Specify the AWS session token to use for the
+            storage.
+        region_name: Specify the S3 region name to use for the storage.
+        s3_endpoint: Specify the S3 endpoint to use for the storage.
+    """
+
+    storage_type: str
+    label_config_type: str
+
+    prefix: Optional[str] = None
+    regex_filter: Optional[str] = ".*"
+    use_blob_urls: Optional[bool] = True
+    presign: Optional[bool] = True
+    presign_ttl: Optional[int] = 1
+    description: Optional[str] = ""
+
+    # credentials specific to the main cloud providers
+    azure_account_name: Optional[str]
+    azure_account_key: Optional[str]
+    google_application_credentials: Optional[str]
+    aws_access_key_id: Optional[str]
+    aws_secret_access_key: Optional[str]
+    aws_session_token: Optional[str]
+    region_name: Optional[str]
+    s3_endpoint: Optional[str]
+
+
+@step(enable_cache=False)
+def get_or_create_dataset(
+    config: LabelStudioDatasetRegistrationConfig,
+    context: StepContext,
+) -> str:
+    """Gets preexisting dataset or creates a new one.
+
+    Args:
+        config: Step config.
+        context: Step context.
+
+    Returns:
+        The dataset name.
+
+    Raises:
+        TypeError: If you are trying to use it with an annotator that is not
+            Label Studio.
+        StackComponentInterfaceError: If no active annotator could be found.
+    """
+    annotator = context.stack.annotator  # type: ignore[union-attr]
+    from zenml.integrations.label_studio.annotators.label_studio_annotator import (
+        LabelStudioAnnotator,
+    )
+
+    if not isinstance(annotator, LabelStudioAnnotator):
+        raise TypeError(
+            "This step can only be used with the Label Studio annotator."
+        )
+    if annotator and annotator._connection_available():
+        preexisting_dataset_list = [
+            dataset
+            for dataset in annotator.get_datasets()
+            if dataset.get_params()["title"] == config.dataset_name
+        ]
+        if (
+            not preexisting_dataset_list
+            and annotator
+            and annotator._connection_available()
+        ):
+            registered_dataset = annotator.register_dataset_for_annotation(
+                config
+            )
+        elif preexisting_dataset_list:
+            return cast(str, preexisting_dataset_list[0].get_params()["title"])
+        else:
+            raise StackComponentInterfaceError("No active annotator.")
+
+        return cast(str, registered_dataset.get_params()["title"])
+    else:
+        raise StackComponentInterfaceError("No active annotator.")
```

Comment:
I think this could be simplified:

```suggestion
    if annotator and annotator._connection_available():
        for dataset in annotator.get_datasets():
            if dataset.get_params()["title"] == config.dataset_name:
                return cast(str, dataset.get_params()["title"])
        
        dataset = annotator.register_dataset_for_annotation(config)
        return cast(str, dataset.get_params()["title"])
        
    raise StackComponentInterfaceError("No active annotator.")
```

Reply:
Thank you. Agreed. I think I sometimes end up with code like this following tussles with `mypy`, but then you're right to then take a step back afterwards and see if what you have at the end really couldn't be simplified down.